File: api/submit.py
import os
import smtplib
from datetime import datetime
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

from flask import Flask, jsonify, request, send_from_directory

# Project root is one level above this file
ROOT_DIR = os.path.join(os.path.dirname(__file__), "..")

app = Flask(__name__)


# ── Static file serving (local dev only — Vercel handles this via CDN) ──
@app.route("/", defaults={"path": ""})
@app.route("/<path:path>")
def static_files(path):
    target = os.path.join(ROOT_DIR, path)
    if path and os.path.exists(target) and os.path.isfile(target):
        directory = os.path.dirname(target)
        filename  = os.path.basename(target)
        return send_from_directory(directory, filename)
    return send_from_directory(ROOT_DIR, "index.html")


# ── POST /api/submit ──
@app.route("/api/submit", methods=["POST"])
def submit():
    data         = request.get_json(silent=True) or {}
    name         = data.get("name", "").strip()
    organisation = data.get("organisation", "").strip()
    communication = data.get("communication", "").strip()

    if not name or not organisation or not communication:
        return jsonify({"error": "Missing required fields"}), 400

    email    = os.environ.get("EMAIL")
    password = os.environ.get("PASSWORD")

    subject = f"SKA Task Submission — {name}"

    body = (
        f"New diagnostic submission received via the Thrive by Design platform.\n\n"
        f"\n"
        f"Name:          {name}\n"
        f"Organisation:  {organisation}\n"
        f"Communication: {communication}\n"
        f"\n\n"
        f"Submitted on: {datetime.now().strftime('%d %b %Y, %I:%M %p')} (PKT)"
    )

    try:
        msg = MIMEMultipart()
        msg["From"]    = f"MUHAMMAD ALI FAROOQ <{email}>"
        msg["To"]      = "work97464@gmail.com"
        msg["Subject"] = subject
        msg.attach(MIMEText(body, "plain"))

        server = smtplib.SMTP("smtp.gmail.com", 587)
        server.starttls()
        server.login(email, password)
        server.sendmail(email, "work97464@gmail.com", msg.as_string())
        server.quit()

        return jsonify({"success": True}), 200

    except Exception as e:
        print(f"Email send error: {e}")
        return jsonify({"error": "Failed to send email"}), 500


# ── Local entry point ──
if __name__ == "__main__":
    from dotenv import load_dotenv
    load_dotenv(os.path.join(ROOT_DIR, ".env"))
    app.run(debug=True, port=3000)

File: api/submit.js
// api/submit.js — Vercel Serverless Function
// Mirrors the Python smtplib/Gmail SMTP approach, implemented in Node.js with Nodemailer.
// Credentials are read from environment variables (never hardcoded).

const nodemailer = require("nodemailer");

module.exports = async function handler(req, res) {
  // Only allow POST
  if (req.method !== "POST") {
    return res.status(405).json({ error: "Method not allowed" });
  }

  const { name, organisation, communication } = req.body;

  // Basic validation
  if (!name || !organisation || !communication) {
    return res.status(400).json({ error: "Missing required fields" });
  }

  // Gmail SMTP transporter — same approach as the Python smtplib code
  const transporter = nodemailer.createTransport({
    service: "gmail",
    auth: {
      user: process.env.EMAIL,      // set in .env or Vercel dashboard
      pass: process.env.PASSWORD,   // 16-char Gmail App Password
    },
  });

  const subject = `SKA Task Submission — ${name}`;

  const body = `
New diagnostic submission received via the Thrive by Design platform.

────────────────────────────────
Name:          ${name}
Organisation:  ${organisation}
Communication: ${communication}
────────────────────────────────

Submitted on: ${new Date().toLocaleString("en-PK", { timeZone: "Asia/Karachi" })} (PKT)
  `.trim();

  try {
    await transporter.sendMail({
      from: `"SKA Diagnostic" <${process.env.EMAIL}>`,
      to: "hello@sarahkhan.co",
      subject,
      text: body,
    });

    return res.status(200).json({ success: true });
  } catch (err) {
    console.error("Email send error:", err.message);
    return res.status(500).json({ error: "Failed to send email" });
  }
};
